### PR TITLE
Compute matter power spectrum

### PR DESCRIPTION
**Pull request by marcelo-alvarez**
Adds ability to compute matter power spectrum.

@1cosmologist note that in order to get the CAMB backend to work I had to comment out `camb_params`. Please have a look at this, thanks.

### CODE REVIEW

File: scripts/example.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import xgcosmo.cosmology   as cosmo
+
+backend = bk.Backend(force_no_gpu=True,force_no_mpi=True)
+cosmo_wsp = cosmo.cosmology(backend, cosmo_backend='CAMB') # for background expansion consistent with websky
+
+k = jnp.logspace(-3,1,1000)
+pk = cosmo_wsp.matter_power(k) # power spectrum
+
+plt.loglog(k, pk, '-')
```
Comment (1cosmologist) [inline scripts/example.py:15]: Adding couple of lines to make plot of the matter power spectrum.

File: xgcosmo/cosmology.py [hunk 2]
```diff
@@ -102,10 +103,16 @@ def __init__(self, backend, **kwargs):
...
+            self.camb_params['H0'] *= 100.
 
             camb_par = camb.set_params(**self.camb_params)
+            camb_par.NonLinear = camb.model.NonLinear_none
+            camb_par.InitPower.set_params(ns=self.params['n_s'])
+            camb_par.set_matter_power(redshifts=[0.,], kmax=2.0)
+
             self.camb_wsp = camb.get_results(camb_par)
+            self._k_grid, z, self._pk = self.camb_wsp.get_matter_power_spectrum(minkh=1e-4, maxkh=1e2, npoints = 2000)
+            self.s8 = jnp.array(self.camb_wsp.get_sigma8())
```
Comment (1cosmologist) [inline xgcosmo/cosmology.py:114]: Do we need s8?

File: xgcosmo/cosmology.py [hunk 3]
```diff
@@ -102,10 +103,16 @@ def __init__(self, backend, **kwargs):
...
-            self.camb_params['HO'] *= 100.
+            self.camb_params['H0'] *= 100.
 
             camb_par = camb.set_params(**self.camb_params)
+            camb_par.NonLinear = camb.model.NonLinear_none
+            camb_par.InitPower.set_params(ns=self.params['n_s'])
+            camb_par.set_matter_power(redshifts=[0.,], kmax=2.0)
+
             self.camb_wsp = camb.get_results(camb_par)
+            self._k_grid, z, self._pk = self.camb_wsp.get_matter_power_spectrum(minkh=1e-4, maxkh=1e2, npoints = 2000)
```
Comment (1cosmologist) [inline xgcosmo/cosmology.py:113]: Perhaps maxkh should go to 1e3 at least. I remember discussing kmax=1e4. Also a grid with more points like the _zgrid?

**Review by 1cosmologist — COMMENTED** [review]
Small changes in example script to make a validation plot for P(k) added. The feature of setting params at least the basic ones seem to work despite the changes. 

One question I have is the choice of kmax for the k_grid, should it be larger? Do we need the sigma8?

**Comment by 1cosmologist** [comment]
Fixed an issue that was causing an error with full `camb_paramset` when using CAMB as backend.

**Comment by marcelo-alvarez** [comment]
Thanks for the fix to the camb backend @1cosmologist. Merging now.

**Pull request merged**